float32[] land_x
float32[] land_y
